backend/shork/src/test/kotlin/compiler/TestParser.kt

```diff
@@ -614,5 +622,353 @@ internal class TestParser {
 
             return arguments
         }
+
+        @JvmStatic
+        fun provideDefaultModifierTests(): List<Arguments> {
+            return provideAllSimpleDefaultModifier() +
+                `provide A Mode immediate, B whatever then AB`() +
+                `provide B-Mode immediate, A-Mode whatever then B`() +
+                `provide neither immediate`() +
+                `provide A not immediate, B whatever then Modifier is B`()
+        }
+
+        // Some instructions always get the same default modifier, no matter the address modes
+        private fun provideAllSimpleDefaultModifier(): List<Arguments> {
+            val inputWithExpected = mutableListOf<Pair<List<Token>, AbstractInstruction>>()
+
+            val firstAddress = 42
+            val secondAddress = 1337
+
+            val instructionWithModifier =
+                listOf(
+                    // DAT and NOP always get F, no matter the address modes
+                    Triple(TokenType.DAT, Dat::class, Modifier.F),
+                    Triple(TokenType.NOP, Nop::class, Modifier.F),
+                    // JMP, JMZ, JMN, DJN, SPL always get B, no matter the address modes
+                    Triple(TokenType.JMP, Jmp::class, Modifier.B),
+                    Triple(TokenType.JMZ, Jmz::class, Modifier.B),
+                    Triple(TokenType.JMN, Jmn::class, Modifier.B),
+                    Triple(TokenType.DJN, Djn::class, Modifier.B),
+                    Triple(TokenType.SPL, Spl::class, Modifier.B),
+                )
+
+            for (instruction in instructionWithModifier) {
+                for (firstAddressMode in addressModes) {
+                    for (secondAddressMode in addressModes) {
```

Comment:
at this point, we might want to think about a `cartesianProduct(list1, list2, list3)` helper or something lol

Reply:
Yeah, something like that might be nice